### PR TITLE
feat(saveUpsert): [INFRA-818] Add schema update for saveUpsert only

### PR DESCRIPTION
**Pull request by Herraj**
## Goal
This PR only contains the gql schema update for **INFRA-818.** Implementation will follow in another PR.

### CODE REVIEW

File: schema.graphql [hunk 1]
```diff
@@ -249,6 +249,25 @@ input SavedItemUpsertInput {
...
 }
+
+"""
+Input field for upserting a Save. Used by saveUpsert mutation
+"""
+input SaveUpsertInput {
+  """
+  The url to create/update the SavedItem with. (the url to save to the list)
+  """
+  url: String!
```
Comment (sri-kirsh) [inline schema.graphql:260]: we need `title` as a part of the input

File: schema.graphql [hunk 2]
```diff
@@ -698,6 +717,16 @@ type Mutation {
...
   ): SaveWriteMutationPayload! @tag(name: "alpha")
+
+  """
+  Creates a new Save; if the Save already exists (either in List or Archive), "re-add" it.
+  "Re-adding" unarchives and undeletes the Save as applicable, and refreshes the "createdAt"
+  timestamp.
+  """
+  saveUpsert(
+  input: [SaveUpsertInput!]! # @constraint(minItems: 1, maxItems: 30),
+  timestamp: ISOString!
```
Comment (sri-kirsh) [inline schema.graphql:728]: this timestamp is already a part of input right?

File: schema.graphql [hunk 3]
```diff
@@ -249,6 +249,25 @@ input SavedItemUpsertInput {
...
+  """
+  url: String!
+  """
+  Optional, create/update the SavedItem as a favorited item
+  """
+  isFavorite: Boolean
+  """
+  Optional, time that request was submitted by client epoch/unix time
+  """
+  timestamp: Int
```
Comment (sri-kirsh) [inline schema.graphql:268]: ok we don't need this timestamp as per the schema proposal:
```
input SaveUpsertInput {
    url: String!
    title: String
    isFavorite: Boolean
  }
```
https://getpocket.atlassian.net/wiki/spaces/PE/pages/2832760905/Accepted+Proposal+Consolidation+V1+Pocket-Graph#Bulk-Mutations

File: schema.graphql (hunk 2 above)
Comment (sri-kirsh) [inline schema.graphql:728]: calling out `timestamp` is here for all saves. we don't need to mention them individually for each SaveUpsertInput https://github.com/Pocket/list-api/pull/701/files#r1149611093

File: schema.graphql [hunk 4]
```diff
@@ -249,6 +250,25 @@ input SavedItemUpsertInput {
...
+  """
+  Optional, title of the SavedItem
+  """
+  title: String
+  """
+  Optional, create/update the SavedItem as a favorited item
+  """
+  isFavorite: Boolean
+}
+
```
Comment (Herraj) [inline schema.graphql:271]: Added this change, rest are just formatting/white space changes.

File: schema.graphql [hunk 5]
```diff
@@ -698,6 +731,16 @@ type Mutation {
...
+  """
+  Creates a new Save; if the Save already exists (either in List or Archive), "re-add" it.
+  "Re-adding" unarchives and undeletes the Save as applicable, and refreshes the "createdAt"
+  timestamp.
+  """
+  saveUpsert(
+    input: [SaveUpsertInput!]! # @constraint(minItems: 1, maxItems: 30),
+    timestamp: ISOString!
+  ): SaveWriteMutationPayload! @tag(name: "alpha")
```
Comment (Herraj) [inline schema.graphql:743]: Added this change, rest are just formatting/white space changes.

**Review by sri-kirsh — APPROVED** [review]

**Pull request merged**